test/src/Number.purs

```diff
@@ -36,6 +36,21 @@ numberTest = describe "Data.Formatter.Number" do
     ["+02.12", "+13.12", "-02.12", "-13.12"]
     (\n →  (format fmt3 <$> (unformat fmt3 n)) `shouldEqual` (Right n))
 
+  forAll (\{fmt: (Formatter fmt), input} -> "rounds up " <> show input <> " (" <> show fmt.after <> " digits)")
+    "rounding"
+    [ {fmt: fmt4, input: 1.99999, expected: "02"}
+    , {fmt: fmt1, input: 1.99999, expected: "002.00"}
+    , {fmt: fmt5, input: 1.99999, expected: "2.0000"}
+    , {fmt: fmt1, input: 1.89999, expected: "001.90"}
+    , {fmt: fmt5, input: 1.67899, expected: "1.6790"}
+    , {fmt: fmt6, input: 12.9, expected: "13"}
```

Comment:
Just to double-check -- there's no issue with large decimals like `1.123456789123456789123456789` / `"1.123456789123456789123456789"`? If that's OK then I'm happy to merge. Sorry for the last question!

Reply:
Oh, yeah no problem! I've added this test and it passes:
```PureScript
    , {fmt: fmt5, input: 1.123456789123456789123456789, expected: "1.1235"}
```
Do you want me to commit that test to this PR?

Reply:
Uh-oh.  I misread what you were asking, and yes, there's a problem if I set `fmt.after` to 27 (like you have in your comment).  I'm looking into it.

Reply:
Okay, so I'm sorry to say that this quickly turned into a mess, and I'm probably going to need some more guidance.  The original implementation has the same limitations that this implementation does, namely:
1. When using `Int.round` (or `floor`), your number gets capped at 2 billion, so we see incorrect values whenever there are more than 9 digits on either side of the decimal point.  E.g., (from the published implementation): 
```PureScript
> format fmtr 12345678901234567890123456789.0
"2,147,483,647.000000000000000002147483647"
> format fmtr 1.23456789012345678901234567890
"1.000000000000000002147483647"
```
2. When a `Number` is greater than or equal to 1e+21, it `show`s in scientific notation, so even if we avoid `Int`s wherever possible, we see very strange values if you have more than 20 digits on either side of the decimal point.

Furthermore, when testing extremely large and small numbers, you often see subtle rounding errors, so it's difficult to assert a specific behavior without hard-coding rounding errors into the test.

I have an implementation that works around these limitations as best as I could come up with, but there are several things I'm unhappy with:
1. I had to change a lot more code than what's in this PR
2. for numbers whose absolute value is >= 1e+21 or <= 1e-21, I append a scientific notation suffix even if `fmt.abbreviations` is `false`, just to avoid giving a blatantly incorrect answer
3. I'm rounding to 20 digits, even if `fmt.after` > 20, again just to avoid giving a blatantly incorrect answer
4. My tests are atrocious since I'm directly asserting rounding errors by running the code in the REPL, and pasting the answer that I observe, e.g.,
```PureScript
    , {fmt: fmt7, input: 1.123456789123456789123456789, expected: "1.123456789123456820000000000"}
```

I could post the implementation that I have, but I guess I'd like to hear first what you think the right path forward is?